doc/source/ray-core/actors/scheduling.rst

```diff
@@ -45,3 +57,11 @@ Since nodes are randomly chosen, actors that don't require any resources are eff
 
         # Zero-CPU (and no other resources) actors are randomly assigned to nodes.
         a2 = Actor.options(num_cpus=0).remote()
+
+        # Only run the actor on the local node.
+        a3 = Actor.options(
+            scheduling_strategy=NodeAffinitySchedulingStrategy(
+                node_id = ray.get_runtime_context().node_id.hex(),
```

Comment:
Before merging this, we wanted to have a standardized example that doesn't need the `.hex()` method here, right? Actually maybe the cleanest solution here is to just omit the `.hex()` here and have `node_id` be of type `NodeID` -- this would parallel our treatment of `ObjectRef` as a Python type.

Reply:
Hmm I agree with you, but we decided on string types for simplicity (see [above conversation](https://github.com/ray-project/ray/pull/23381#issuecomment-1085387958)).

It would be easy to keep backwards compatibility for string types vs NodeIDs, but I agree compatibility will be more annoying for the `.hex()` call so it's more straightforward for users to just use whatever type is returned by the existing ray APIs. What do you think, @ericl?

Reply:
We do want to change runtime_context to return node_id as hex string: https://github.com/ray-project/ray/pull/23715 but that's backward incompatible and people think we should do it for Ray 2.0. 

What I can do here is changing NodeAffinitySchedulingStrategy to accept both NodeID and string so we won't see `.hex()` in the user doc and in Ray 2.0 drop the support for NodeID. WDYT?

Reply:
Ah didn't realize we were going to drop user-facing NodeID anyway. In that case I like @jjyao's plan to accept both for now so it's easy to deprecate NodeID later!

Reply:
I'm fine with using a string, but then the user should only ever see the hex version of the string and there should be no other public APIs the return NodeID or the binary node id.

This is the first time we introduce a public API dependency on the node id, so I think we should do it as part of this PR and not introduce this `.hex()` thing now and then change it later for Ray 2.0. Just be very precise what the public API is on getting node ids and how they can be passed to `NodeAffinitySchedulingStrategy`.

Reply:
The issue is that we already have a (and the only) public api that returns NodeID (i.g. runtime_context) and I think we shouldn't change that API in this PR.

With my proposal, users don't need to call `.hex()`, they can directly use the node id returned by runtime_context. Later when we change runtime_context to return string instead of NodeID, the same code will still work so it's transparent to the users.

We can also make this api private (no user-facing document) for now until we have public apis to get string node id.

Reply:
The plan to accept both for now and then migrate for Ray 2.0 sounds good to me!